libraries/provider_td_rubygems.rb

```diff
@@ -40,6 +43,39 @@ def td_plugin_name
           "fluent-plugin-#{@new_resource.package_name}" 
         end
 
+        def rubygems_version
```

Comment:
I think we don't have to have this method definition if we use `TdGemEnvironment` properly 🤔  I couldn't get your intention to define multiple `rubygems_version` in multiple places.